### PR TITLE
PBL-32106 GBitmap

### PR DESCRIPTION
**Pull request by HBehrens**
- tests will fail as I didn't push an updated `transpiled.js`, yet
- also missing: `.onload` and `.onerror` callbacks on `GBitmap`.

### CODE REVIEW

**Comment by HBehrens** [comment]
ping @franc0is

**Comment by franc0is** [comment]
Still missing: gbitmap_get/set_palette. We'll have to figure out what to do with the array of gcolor.

For set_palette, I think we just need to create a JS array of GColor and pass it to the cwrap'ed function with 'array' argument specified. As far as I understand, emscripten assumes it is a uint8 array.

For get_palette, we need to create the JS array from the pointer.

In both cases, we need gbitmap_get_palette_size, so I added that too. However, calling get_format->get_palette_size->set_palette is quite cumbersome so we might want to tweak that API a bit

**Comment by franc0is** [comment]
Also TODO: unit tests for the new APIs

**Comment by HBehrens** [comment]
@franc0is I am not sure we need palette handling at all for our JS APIs. It's a rather uncommon concept in the Canvas world. I will wrap up the tests for the remaining functions this morning.

File: examples/simple/index.html [hunk 1]
```diff
@@ -86,9 +88,36 @@
         // the same way you were using them in C
         rocky.export_global_c_symbols();
 
+        // note: the proxy URL is subject to change and will probably baked into the gbitmap_create()
```
Comment (franc0is) [inline examples/simple/index.html]: will probably *be* baked.

Comment (HBehrens) [inline examples/simple/index.html]: :white_check_mark:

**Comment by franc0is** [comment]
:+1:

**Pull request merged**